pkg/eks/client_test.go

```diff
@@ -104,8 +147,8 @@ var _ = Describe("eks auth helpers", func() {
 				})
 
 				It("should create config with embedded token", func() {
-					// TODO: cannot test this, as token generator uses STS directly, we cannot pass the interface
-					// we can probably fix the package itself
+					testClientConfig("")
+					testClientConfig("arn:aws:iam::111111111111:role/eksctl")
```

Comment:
might be worth just inling the assertions rather than having this function call. Its quite hard to read the test otherwise. Wdyt?